### ISSUE TITLE
runserver_plus creates two pin codes for werkzeug debugger

### ISSUE THREAD

**Issue by Apreche** [django-extensions/django-extensions#832]
Whenever I use runserver_plus to start a development server using the werkzeug debugger, it activates the debugger twice and prints out two pin codes. I expect there to be only one. Other than printing two pin codes in the terminal, the server runs perfectly as expected.

Steps to replicate:

```
# create a new empty virtualenv
pip install django ==1.9.2
pip install django-extensions==1.6.1
pip install werkzeug==0.11.3
django-admin.py startproject testproj
# Edit the testproj INSTALLED_APPS setting to include django_extensions
./manage.py runserver_plus
```

Output looks like this.
```
...
 * Debugger is active!
 * Debugger pin code: 140-764-085
 * Debugger is active!
 * Debugger pin code: 895-622-303
```

**Comment by tony** [comment]
This has been happening to me since forever. And its messin my jive up.

I'm about to just dig into the code and figure it out.

Anyone have something to add?

**Comment by kkroboth** [comment]
Have same problem as well. Appears to be how werkzeug handles reloading. I think it starts a new process.
https://github.com/mitsuhiko/werkzeug/blob/master/werkzeug/serving.py#L658
https://github.com/mitsuhiko/werkzeug/blob/master/werkzeug/_reloader.py#L90

**Comment by tony** [comment]
Yep that makes sense. This happens in Flask too.

https://stackoverflow.com/questions/9449101/how-to-stop-flask-from-initialising-twice-in-debug-mode

We need to make this stop happening.

The ``runserver_plus`` command we use in [runserver_plus.py](https://github.com/django-extensions/django-extensions/blob/master/django_extensions/management/commands/runserver_plus.py) ([history](https://github.com/django-extensions/django-extensions/commits/master/django_extensions/management/commands/runserver_plus.py)) is still modeled after [serving.py](https://github.com/mitsuhiko/werkzeug/blob/master/werkzeug/serving.py) ([history](https://github.com/mitsuhiko/werkzeug/commits/master/werkzeug/serving.py)) in werkzeug.

**Comment by tony** [comment]
``inner_run`` will end up being called twice.

if you run ``DebuggedApplication`` only on the outer-loop, you preserve reloading when files are changed as well as only have to worry about once instance of ``DebuggedApplication``.

This is a bit counter-intuitive, I was expecting ``DebuggedApplication`` to want to be in the inner loop, when ``WERKZEUG_RUN_MAIN`` is ``'true'`` (yes, that's a string with single quotes). However, when I tried that (along with various combinations of ``use_debugger`` and ``use_reloader`` depending on the ``WERKZEUG_RUN_MAIN``, the ability reload functionality would be lost.

In any event, I would appreciate if you could QA my branch on your end, @kkroboth , @Apreche, anyone else out there.

```sh
pip install -e git+https://github.com/tony/django-extensions@runserver-plus-doublerun#egg=django-extensions
```

**Comment by tony** [comment]
Hey, can any of you guys give it a shot?

**Comment by Apreche** [comment]
@tony I tried your fork out. Now when I launch the runserver_plus I get no pin-codes whatsoever. 

```
$ ./manage.py runserver_plus 0.0.0.0:8000
 * Running on http://0.0.0.0:8000/ (Press CTRL+C to quit)
 * Restarting with stat
Performing system checks...

System check identified no issues (0 silenced).

Django version 1.7.7, using settings 'myproject.settings'
Development server is running at http://0.0.0.0:8000/
Using the Werkzeug debugger (http://werkzeug.pocoo.org/)
Quit the server with CONTROL-C.
```

**Comment by kkroboth** [comment]
@tony There is a [misplaced comma](https://github.com/tony/django-extensions/commit/10fa1aca273f6bdb6547e03220893480f659f8e2#diff-3d83f4d549c1e3b62522a07d357cea88R344) making a tuple for handler.

**With reload**
```
$ ./manage.py runserver_plus 

 * Running on http://127.0.0.1:8000/ (Press CTRL+C to quit)
 * Restarting with stat

Performing system checks...

System check identified no issues (0 silenced).

Django version 1.9.2, using settings 'testproj.settings'
Development server is running at http://127.0.0.1:8000/
Using the Werkzeug debugger (http://werkzeug.pocoo.org/)
Quit the server with CONTROL-C.
 * Debugger is active!
 * Debugger pin code: 423-442-950
```

**Without reload**
Works after removing comma.
```
$ ./manage.py runserver_plus --noreload

 * Running on http://127.0.0.1:8000/ (Press CTRL+C to quit)
```
The debuggers work in both instances, but the code is only shown after console is opened in browser if using *--noreload*.

**Comment by tony** [comment]
@kkroboth Sorry for the delay.

Rebased and fixed comma issue.

**Comment by tony** [comment]
@kkroboth In addition, fixed the issue with ``--noreload`` @ 4d83167.

I was able to test the werkzeug repl with and without ``--noreload``.

**Comment by trbs** [comment]
Merged @tony 's PR into master.

Everybody please test ```runserver_plus``` in master.
Thanks !
